projects/hipblaslt/tensilelite/Tensile/Tests/unit/test_VerifyGRsCompleteBeforeLr1s.py

```diff
@@ -0,0 +1,386 @@
+        assert message == "Code path 0: GRA at index 0 is not valid. There are no guarantees on when it will be done."
+
+    def test_no_sbarrier(self): 
+        """
+        Failure case where there is no SWait between GRs and LR1s.
+        """
+        assert self.num_vmfma == 8
+
+        optSchedule = {
+            "SYNC": [[0, 1, self.num_vmfma-1]],
+            "LRA0": [[-1]],
+            "LRB0": [[-1]],
+            "GRA":  [[0]],
+            "GRB":  [[0]],
+            "LRA1": [[6]],
+            "LRB1": [[6]]
+        }
+
+        syncCode = [
+            SWaitCnt(dscnt=0, vlcnt=-1, vscnt=-1, comment="Wait for LR0s"),
+            SWaitCnt(dscnt=-1, vlcnt=2, vscnt=-1, comment="Wait for GRs"),
+            SWaitCnt(dscnt=0, vlcnt=-1, vscnt=-1, comment="Wait for LR1s"),
+        ]
+        sched = ScheduleInfo(1, self.num_vmfma, optSchedule, syncCode, None, None)
+        status, message = verify_grs_complete_before_lr1s(sched, {"kernel": self.kernel})
+        assert not status, f"Schedule should have failed (GRA is not guaranteed before LRA1), but passed. {message}"
+        assert message == "Code path 0: GRA at index 0 is not valid. There is no SBarrier acting on it."
+
+    def test_swait_after_sbarrier(self):
+        """
+        Failure case when the latest barrier acting on the GRs occurs BEFORE the SWaitCnt that guarantees those GRs.
+        """
+        assert self.num_vmfma == 8
+
+        optSchedule = {
+            "SYNC": [[0, 2, 3, self.num_vmfma-1]],
+            "LRA0": [[-1]],
+            "LRB0": [[-1]],
+            "GRA":  [[0]],
+            "GRB":  [[0]],
+            "LRA1": [[6]],
+            "LRB1": [[6]]
+        }
+
+        syncCode = [
+            SWaitCnt(dscnt=0, vlcnt=-1, vscnt=-1, comment="Wait for LR0s"),
+            SBarrier(comment="For GRs"),
+            SWaitCnt(dscnt=-1, vlcnt=2, vscnt=-1, comment="Wait for GRs"),
+            SWaitCnt(dscnt=0, vlcnt=-1, vscnt=-1, comment="Wait for LR1s"),
+        ]
+        sched = ScheduleInfo(1, self.num_vmfma, optSchedule, syncCode, None, None)
+        status, message = verify_grs_complete_before_lr1s(sched, {"kernel": self.kernel})
+        assert not status, f"Schedule should have failed (no barrier between GRA and LRA1), but passed. {message}"
+        assert message == "Code path 0: GRA at index 0 is not valid. No SBarrier between SWait @ idx=3 and LR1 @ idx=6. Order must be GR > SWait > SBarrier > LR1."
+
+    def test_guaranteed_after_first_lr1(self):
+        """
+        Simple failure case where the last GR is guaranteed AFTER the first LR1 starts.
+        """
+        assert self.num_vmfma == 8
+
+        optSchedule = {
+            "SYNC": [[0, 4, 4, self.num_vmfma-1]],
+            "LRA0": [[-1]],
+            "LRB0": [[-1]],
+            "GRA":  [[0,0,0,0]],
+            "GRB":  [[0,0,0,0]],
+            "LRA1": [[3,6,6,6]],
+            "LRB1": [[3,6,6,6]]
```

Comment:
```suggestion
            "LRB1": [[6,6,6,6]]
```

Reducing the number of errors that could be thrown makes it a bit easier to understand the test, and makes the test a bit less brittle (to someone permuting the logic in the rule)